pkg/pillar/cmd/volumemgr/updatestatus.go

```diff
@@ -83,6 +85,20 @@ func doUpdate(ctx *volumemgrContext, status *types.VolumeStatus) (bool, bool) {
 		// Asynch creation; ensure we have requested it
 		MaybeAddWorkCreate(ctx, status)
 	}
+	// If container image is already available in verified directory
+	// then flow willt not be through the downloader. So, we are looking
+	// for size in volume status. If it is 0 then we are updating it.
+	if status.DownloadOrigin.MaxSizeBytes == 0 &&
+		status.DownloadOrigin.IsContainer {
+
+		info, err := os.Stat(status.FileLocation)
+		if err != nil {
+			errStr := fmt.Sprintf("Calculating size of container image failed: %v", err)
+			log.Error(errStr)
+		} else {
+			status.DownloadOrigin.MaxSizeBytes = uint64(info.Size())
```

Comment:
This is not the current size

Reply:
We are setting up the download size for containers here because it is 0 in config.

Reply:
I don't want container-specific code. This adds more container specific code.

Reply:
We are not doing it separately. At the time of volume creation, if the initial status is available then we are preparing the snapshot again in volumemgr. So, at the same time we are also setting up the downloaded size also because the size is zero in the volume config from the zedmanager for the container applications.

Reply:
My point is that we should not add any code in a container/isContainer specific path.
We need to make the code more uniform and this makes it less uniform across VMs and containers.

Reply:
Thanks